test/src/specs/rpc/submit_block.rs

```diff
@@ -0,0 +1,118 @@
+
+        // build block with wrong block number
+        let block = node0
+            .new_block_builder(None, None, None)
+            .number(2.pack())
+            .build();
+        let block_err = node0
+            .rpc_client()
+            .submit_block("".to_owned(), block.data().into())
+            .unwrap_err();
+        assert!(
+            block_err.to_string().contains("NumberError"),
+            "Submit block with wrong block number should return 'NumberError'"
+        );
+
+        // build block with wrong block version
+        let block = node0
+            .new_block_builder(None, None, None)
+            .version((42 as u32).pack())
+            .build();
+        let block_err = node0
+            .rpc_client()
+            .submit_block("".to_owned(), block.data().into())
+            .unwrap_err();
+        assert!(
+            block_err.to_string().contains("BlockVersionError"),
+            "Submit block with wrong block version should return 'BlockVersionError'"
+        );
+
+        // build block with wrong epoch
+        let block = node0
+            .new_block_builder(None, None, None)
+            .epoch(42.pack())
+            .build();
+        let block_err = node0
+            .rpc_client()
+            .submit_block("".to_owned(), block.data().into())
+            .unwrap_err();
+        assert!(
+            block_err.to_string().contains("Epoch(NumberMismatch"),
+            "Submit block with wrong epoch should return 'EpochNumberMismatch"
+        );
+
+        // build block with wrong parent_hash: ancestor block hash
+        let ancestor_block_hash = node0.get_block_by_number(1).hash();
+        let block = node0
+            .new_block_builder(None, None, None)
+            .parent_hash(ancestor_block_hash)
+            .build();
+        let block_err = node0
+            .rpc_client()
+            .submit_block("".to_owned(), block.data().into())
+            .unwrap_err();
+        assert!(
+            block_err.to_string().contains("NumberError"),
+            "Submit block with parent hash point to an ancestor should return 'NumberError'"
+        );
+
+        // build block with wrong parent_hash: unknown block hash
+        let another_tip_block_hash = node1.get_tip_block().hash();
+        let block = node0
+            .new_block_builder(None, None, None)
+            .parent_hash(another_tip_block_hash)
+            .build();
+        let block_err = node0
+            .rpc_client()
+            .submit_block("".to_owned(), block.data().into())
+            .unwrap_err();
+        assert!(
+            block_err.to_string().contains("UnknownParentError"),
+            "Submit block with wrong parent hash should return 'UnknownParentError'"
+        );
+
+        // build block with wrong timestamp: block time too new
+        let block = node0
+            .new_block_builder(None, None, None)
+            .timestamp((now_ms() + 16_000).pack())
```

Comment:
Add some comments around `16_000` and the relation with `ALLOWED_FUTURE_BLOCKTIME`.
And suggest `30s` as the ci machine is poor.

Reply:
I'm going to add comment about this together with updating epoch relate functions after #2495